presto-hive/src/main/java/com/facebook/presto/twitter/hive/thrift/HiveThriftFieldIdResolver.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.twitter.hive.thrift;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.airlift.log.Logger;
+
+import java.util.Properties;
+
+public class HiveThriftFieldIdResolver
+        implements ThriftFieldIdResolver
+{
+    private static final Logger log = Logger.get(HiveThriftFieldIdResolver.class);
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+    public static final String THRIFT_FIELD_ID_JSON = "thrift.field.id.json";
+    private final JsonNode root;
+
+    public HiveThriftFieldIdResolver()
+    {
+        this.root = null;
+    }
+
+    public HiveThriftFieldIdResolver(JsonNode root)
+    {
+        this.root = root;
+    }
+
+    public ThriftFieldIdResolver initialize(Properties schema)
```

Comment:
I think the interface/impl pattern is fine, just don't mix in the builder pattern as well via instance methods on the impl. The interface should just have what's needed for their usage (i.e, `short getThriftId(int hiveIndex);`). The builder methods should not be included and should be handled separately as static initializes. When changing the implementation, a new factory method on a new class can be used, but the downstream consumers would remain unaffected since the interface contract they care about is consistent.